src/python/WMArchive/PySpark/RecordAggregator.py

```diff
@@ -17,10 +18,100 @@ def get_scope_hash(scope):
     return str(hash(frozenset(scope.items())))
 
 
+def extract_stats(record):
```

Comment:
Nils, please add to every function definition its docstring. We'll generate documentation out of docstring. Please be expressive and describe in detail what functions are doing.